src/hw_tools.py

```
@@ -175,6 +176,44 @@ def remove(self) -> None:
         # the remove option.
 
 
+class SnapStrategy(StrategyABC):
+    """Snap strategy class."""
+
+    def __init__(self, tool: HWTool, channel: str):
+        """Snap strategy constructor."""
+        self._name = tool
+        self.snap_name = tool.value
+        self.channel = channel
+        self.snap_client = snap.SnapCache()[tool.value]
+
+    def install(self, channel: str = "latest/stable") -> None:
+        """Install the snap from a channel."""
+        try:
+            snap.add(self.snap_name, channel=channel)
+        # using the snap.SnapError will result into:
+        # TypeError: catching classes that do not inherit from BaseException is not allowed
+        except Exception as err:  # pylint: disable=broad-except
+            logger.error(
+                "Failed to install %s from channel: %s: %s", self.snap_name, self.channel, err
+            )
```

This logs an error to the charm log, but that's silent to the user - can we bubble this error up to the charm status?  How do the other exporters handle installation failures?

---

Forgot to re-raise the error. Thanks